ambry-rest/src/main/java/com.github.ambry.rest/NettyServer.java

```diff
@@ -62,6 +62,7 @@ public NettyServer(NettyConfig nettyConfig, NettyMetrics nettyMetrics,
     this.nettyMetrics = nettyMetrics;
     this.channelInitializer = channelInitializer;
     NettyRequest.bufferWatermark = nettyConfig.nettyServerRequestBufferWatermark;
+    NettyRequest.maxBytesPerRead = nettyConfig.nettyServerMaxBytesPerRead;
```

Comment:
4.1.4 has eliminated the [API](https://netty.io/4.0/api/io/netty/channel/ChannelConfig.html#setMaxMessagesPerRead(int)) to set max messages to read. Instead, it has introduced the concept of a ["receive buffer allocator"](https://netty.io/4.1/api/io/netty/channel/ChannelConfig.html#setRecvByteBufAllocator(io.netty.channel.RecvByteBufAllocator)). One of these [implementations](https://netty.io/4.1/api/io/netty/channel/DefaultMaxBytesRecvByteBufAllocator.html) seems useful to us and it takes as argument the number of bytes to read on every read loop.

Previously, we were setting max messages to read to 1 and no configuration was required. With this implementation, the number of bytes that should be read can be set - so I thought that making it configurable is a good idea. I realize that the user might not have enough information to set this - do you think we should just set this to a preconfigured number?